File: apps/api/src/modules/timeline/timeline.controller.ts
import { Controller, Get, Inject, Query } from "@nestjs/common";
import { ApiRoutes, type TimelinePageDto } from "@today/contracts";
import { TimelineService } from "./timeline.service";

@Controller()
export class TimelineController {
  constructor(
    @Inject(TimelineService) private readonly timeline: TimelineService,
  ) {}

  @Get(ApiRoutes.timeline.replace(/^\//, ""))
  list(@Query("limit") limit?: string): TimelinePageDto {
    const n = limit ? Number(limit) : 30;
    return this.timeline.list(Number.isFinite(n) ? n : 30);
  }
}

File: apps/api/src/main/java/com/today/timeline/TimelineController.java
package com.today.timeline;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TimelineController {

  private final TimelineService timeline;

  public TimelineController(TimelineService timeline) {
    this.timeline = timeline;
  }

  @GetMapping("/v1/timeline")
  public TimelinePageDto list(@RequestParam(required = false) Integer limit) {
    int n = limit != null && limit > 0 ? limit : 30;
    return timeline.list(n);
  }
}
